### PR TITLE
Enable GitHub CodeQL code scanning

### PR DESCRIPTION
**Pull request by briandealwis**
Earn another star on the [OSSF scorecard](https://github.com/ossf/scorecard/blob/main/checks/checks.md#sast).

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    # ℹ️ Command-line programs to run using the OS shell.
+    # 📚 https://git.io/JvXDl
+
+    # ✏️ If the Autobuild fails above, remove it and uncomment the following three lines
+    #    and modify them (or add more) to build your code if your project
+    #    uses a compiled language
+
+    #- run: |
+    #   make bootstrap
+    #   make release
```
Comment (MarlonGamez) [inline .github/workflows/codeql-analysis.yml:59]: Looks like we'll have to replace this with just a `make` for our repo?

Comment (briandealwis) [reply .github/workflows/codeql-analysis.yml:59]: I should have made this a draft! 😊

**Comment by briandealwis** [comment]
Oh jeepers, this takes 35 minutes.

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    branches: [ master ]
+  schedule:
+    - cron: '23 17 * * 3'
+
+permissions: read-all
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
```
Comment (nkubala) [inline .github/workflows/codeql-analysis.yml:17]: do we want this to run on other platforms too? I guess if it's running `make` then our hands are kinda tied 🤷

**Review by nkubala — APPROVED** [review]

**Comment by nkubala** [comment]
> Oh jeepers, this takes 35 minutes.

well....so do our integration tests 😭 and since this isn't running on the same platform I don't think it will make average CI times any longer.

**Comment by briandealwis** [comment]
May be better to run this as a nightly?

**Comment by tejal29** [comment]
> Earn another star on the [OSSF scorecard](https://github.com/ossf/scorecard/blob/main/checks/checks.md#sast).

💯 ⭐ ⭐ 🌟

File: .github/workflows/codeql-analysis.yml (hunk 2 above)
Comment (briandealwis) [reply .github/workflows/codeql-analysis.yml:17]: I don't know how this works exactly, but as a code-scanner it shouldn't much matter.  I wonder if it extracts paths from the debugging symbols.

**Comment by verbanicm** [comment]
@briandealwis Do you need any help getting this finished?

**Comment by briandealwis** [comment]
Thanks for the bump @verbanicm: I think the "blocker" here is deciding when to run these tests.  I'll set it up a nightly (which I think means I'll be the only one to receive notifications for now) and let's see how it goes.

**Comment by briandealwis** [comment]
```
Analysis produced the following diagnostic data:

| Severity |                          Message                           |
+----------+------------------------------------------------------------+
| none     | (659 results for diagnostic "Successfully analyzed files") |
Analysis produced the following metric data:

|                 Metric                 | Value |
+----------------------------------------+-------+
| Total lines of Go code in the database | 66344 |

Counted a baseline of 345461 lines of code for go.
```

**Pull request merged**